mentat/commands.py

```diff
@@ -19,11 +19,21 @@ def __init_subclass__(cls, command_name: str | None) -> None:
             Command._registered_commands[command_name] = cls
 
     @classmethod
-    def create_command(cls, command_name: str) -> Command:
+    def create_command(
+        cls, command_name: str, code_context: Optional[CodeContext] = None
+    ) -> Command:
         if command_name not in cls._registered_commands:
             return InvalidCommand(command_name)
+
+        command_cls = cls._registered_commands[command_name]
+        if command_cls in [AddCommand, RemoveCommand]:
```

Comment:
Hmm, at this point why not just have code_context be an instance variable of the Command class so every command has access to it?

Reply:
Eventually we'll come up with a cleaner way of sharing global singletons like CodeContext. First we need to see how the interface refactor goes which will inform whether we might eventually have multiple Mentat conversations running on the same process or not.

Until then I think this is fine, it makes it easier to read `AddCommand` and `RemoveCommand` and understand what they have access to.